language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -0,0 +1,620 @@
+        String returnStatement = "";
+        if (updatingVar.isPresent() && updatingVar.get().getName().isPresent()) {
+            returnStatement = String.format("return %s;", updatingVar.get().getName().get());
+        }
+
+        List<Node> selectedNodes = statementAnalyzer.getSelectedNodes();
+        String funcBody = selectedNodes.stream().map(Node::toSourceCode).collect(Collectors.joining(""));
+        Range extractFunctionInsertRange;
+
+        boolean newLineAtEnd = addNewLineAtEnd(enclosingNode.get());
+        extractFunctionInsertRange = PositionUtil.toRange(enclosingNode.get().lineRange().endLine());
+
+        // Check if the function call is invoked from an isolated context.
+        IsolatedBlockResolver isolatedBlockResolver = new IsolatedBlockResolver();
+        Boolean isIsolated = isolatedBlockResolver.findIsolatedBlock(matchedCodeActionNode);
+
+        String extractFunction = FunctionGenerator.generateFunction(functionName, argsForExtractFunction,
+                returnTypeDescriptor, returnStatement, newLineAtEnd, isIsolated, funcBody);
+
+        try {
+            // Adding a line separator which is removed by the Formatter
+            extractFunction = CommonUtil.LINE_SEPARATOR + Formatter.format(extractFunction).stripTrailing();
+        } catch (FormatterException e) {
+            return Collections.emptyList();
+        }
+        String replaceFunctionCall = getReplaceFunctionCall(argsForReplaceFunctionCall, functionName, false);
+
+        if (updatingVar.isPresent() && updatingVar.get().getName().isPresent()) {
+            String varName = updatingVar.get().getName().get();
+            replaceFunctionCall =
+                    String.format("%s %s = %s", possibleTypeOfUpdatingVar.get(), varName, replaceFunctionCall);
+        } else if (matchedCodeActionNode.kind() == SyntaxKind.RETURN_STATEMENT) {
+            replaceFunctionCall = String.format("return %s", replaceFunctionCall);
+        }
+
+        Position replaceFuncCallStartPos = PositionUtil.toPosition(selectedNodes.get(0).lineRange().startLine());
+        Position replaceFuncCallEndPos = PositionUtil
+                .toPosition(selectedNodes.get(selectedNodes.size() - 1).lineRange().endLine());
+        Range replaceFuncCallInsertRange = new Range(replaceFuncCallStartPos, replaceFuncCallEndPos);
+
+        TextEdit extractFunctionEdit = new TextEdit(extractFunctionInsertRange, extractFunction);
+        TextEdit replaceFunctionCallEdit = new TextEdit(replaceFuncCallInsertRange, replaceFunctionCall);
+
+        CodeAction codeAction = CodeActionUtil.createCodeAction(CommandConstants.EXTRACT_TO_FUNCTION,
+                List.of(extractFunctionEdit, replaceFunctionCallEdit), context.fileUri(),
+                CodeActionKind.RefactorExtract);
+
+        return List.of(codeAction);
+    }
+
+    /**
+     * Finds whether the selected range is extractable and if yes, whether a value should be returned or if yes,
```

Comment:
Why do we have `and if yes` and `or if yes` in the same statement?

Reply:
Will simplify to `Finds whether the selected range is extractable. If extractable, finds whether a value should be returned or not from the extracted function. If yes, return the updated variable symbol.` WDYT?

Reply:
+1